django_redis/hash_ring.py

```diff
@@ -1,19 +1,22 @@
 import bisect
 import hashlib
+from typing import Dict, Generator, List, Optional, Tuple, Union
 
 
 class HashRing:
-    nodes = []
+    nodes = []  # type: List[str]
 
-    def __init__(self, nodes=(), replicas=128):
-        self.replicas = replicas
-        self.ring = {}
-        self.sorted_keys = []
+    def __init__(
+        self, nodes: Union[List[str], Tuple[str, ...]] = (), replicas: int = 128
```

Comment:
This is probably more appropriate because the order is respected but you can pass any iterable (or `Sequence[str]`).

```suggestion
        self, nodes: Iterable[str] = (), replicas: int = 128
```

Reply:
I did not know `Iterable` could be used for tuple as well. I usually prefer to have a more specific type but I can't argue with simplicity